server/src/test/java/org/cloudfoundry/identity/uaa/scim/jdbc/JdbcScimUserProvisioningTests.java

```diff
@@ -443,7 +447,11 @@ void retrieveByScimFilter_IncludeInactive() {
             );
             Assertions.assertThat(result).isNotNull();
             final List<String> usernames = result.stream().map(ScimUser::getUserName).toList();
-            Assertions.assertThat(usernames).isSorted();
```

Comment:
Isn't it a difference of functionality with different databases. Is it something that needs to be fixed?

Reply:
I looked into it. The test calls:

```java
jdbcScimUserProvisioning.query(
                    ...,
                    "userName",
                    ...,
                    ...
            )
```

So it sorts by username alphabetically, whatever "alphabetical" means for the given database. In production code it's only ever sorted by `created`.

I updated the test to reflect this.

Reply:
hi @coolgang123 

Yes, the databases indeed are different, but it also depends on locale for the sorting. So the test itself is not all that meaningful, because the sort order can change when the database configuration changes.

If we wanted to test all DBs we could use the following fix, but I would still vote to remove the check for MySQL 

```
diff --git a/server/src/test/java/org/cloudfoundry/identity/uaa/scim/jdbc/JdbcScimUserProvisioningTests.java b/server/src/test/java/org/cloudfoundry/identity/uaa/scim/jdbc/JdbcScimUserProvisioningTests.java
index 864f1af2c..9c1d0b7b6 100644
--- a/server/src/test/java/org/cloudfoundry/identity/uaa/scim/jdbc/JdbcScimUserProvisioningTests.java
+++ b/server/src/test/java/org/cloudfoundry/identity/uaa/scim/jdbc/JdbcScimUserProvisioningTests.java
@@ -448,9 +448,11 @@ class JdbcScimUserProvisioningTests {
             Assertions.assertThat(result).isNotNull();
             final List<String> usernames = result.stream().map(ScimUser::getUserName).toList();
             if (Arrays.stream(enviroment.getActiveProfiles()).noneMatch("mysql"::equalsIgnoreCase)) {
-                // MySQL has a different ordering from HSQL and Postgres. In MySQL the two values
-                // we inserted are sorted in reverse order, so we skip the assert entirely
                 Assertions.assertThat(usernames).isSorted();
+            } else {
+                //MySQL has a different ordering from HSQL and Postgres. In MySQL the two values
+                //we inserted are sorted in reverse order,
+                Assertions.assertThat(usernames).isSortedAccordingTo((o1, o2) -> o2.compareTo(o1));
             }
             return usernames;
         };
```

Reply:
sounds good